v-next/hardhat/src/internal/builtin-plugins/solidity-test/helpers.ts

```diff
@@ -0,0 +1,88 @@
+import type { ArtifactsManager } from "../../../types/artifacts.js";
+import type {
+  ArtifactId,
+  SuiteResult,
+  Artifact,
+  SolidityTestRunnerConfigArgs,
+  TestResult,
+} from "@ignored/edr";
+
+import { runSolidityTests } from "@ignored/edr";
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+
+/**
+ * Run all the given solidity tests and returns the whole results after finishing.
+ */
+export async function runAllSolidityTests(
```

Comment:
Both the functions in this file are direct ports of Franco's v2 plugin and its supporting code: https://github.com/NomicFoundation/edr/blob/feat/solidity-tests/js/helpers/src/index.ts

I have no commitment to this structuring and I am happy to see it refactored.

Reply:
Could we add that as a comment, too?

Reply:
I have added a comment.